### PR TITLE
podmanv2 checkpoint and restore

### PR DESCRIPTION
**Pull request by baude**
add the ability to checkpoint and restore containers on v2podman

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/libpod/containers.go [hunk 1]
```diff
@@ -325,3 +326,84 @@ func ListContainerBatch(rt *libpod.Runtime, ctr *libpod.Container, opts shared.P
...
 	return ps, nil
 }
+
+func Checkpoint(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Keep           bool   `schema:"keep"`
+		LeaveRunning   bool   `schema:"leaveRunning"`
+		TCPEstablished bool   `schema:"tcpEstablished"`
+		Export         string `schema:"export"`
```
Comment (mheon) [inline pkg/api/handlers/libpod/containers.go]: We can't just pass this over - we'll be creating the export tar on the server. Expected behavior would be the client, I imagine. If export is requested we probably need to save the tar to a temporary file and stream it back to the client.

File: pkg/api/handlers/libpod/containers.go [hunk 2]
```diff
@@ -325,3 +326,84 @@ func ListContainerBatch(rt *libpod.Runtime, ctr *libpod.Container, opts shared.P
...
+	}
+	utils.WriteResponse(w, http.StatusOK, entities.CheckpointReport{Id: ctr.ID()})
+}
+
+func Restore(w http.ResponseWriter, r *http.Request) {
+	decoder := r.Context().Value("decoder").(*schema.Decoder)
+	query := struct {
+		Keep            bool   `schema:"keep"`
+		TCPEstablished  bool   `schema:"tcpEstablished"`
+		Import          string `schema:"import"`
```
Comment (mheon) [inline pkg/api/handlers/libpod/containers.go]: Same as Export, but in reverse - the tar is going to be on the client, so we need to stream it to the server, save to a temporary file, and then pass that into libpod Restore.

**Comment by mheon** [comment]
Suggestion: strip import/export and do them in a separate PR, rest is probably ready to merge as is.

**Comment by baude** [comment]
ready for merge

File: pkg/api/handlers/libpod/containers.go [hunk 3]
```diff
@@ -325,3 +330,129 @@ func ListContainerBatch(rt *libpod.Runtime, ctr *libpod.Container, opts shared.P
 	}
 	return ps, nil
 }
+
+func Checkpoint(w http.ResponseWriter, r *http.Request) {
```
Comment (mheon) [inline pkg/api/handlers/libpod/containers.go:334]: Needs doc comment

Comment (mheon) [reply pkg/api/handlers/libpod/containers.go:334]: Can wait till later, though, given this is just an HTTP endpoint

**Comment by mheon** [comment]
LGTM

File: pkg/api/handlers/libpod/containers.go (hunk 3 above)
Comment (baude) [reply pkg/api/handlers/libpod/containers.go:334]: none of the endpoints have doc comments

**Comment by rhatdan** [comment]
I want @adrianreber to make sure this works but It LGTM now. Just so we can get to testing.
/lgtm

**Pull request merged**